File: engine/engine-model/src/main/scala/tools/packet/alliance/AllianceInvite.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class AllianceInvite(private var _allianceId: Int, private var _characterName: String) extends PacketInput {
  def allianceId: Int = _allianceId

  def characterName: String = _characterName

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/AllianceInvite.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record AllianceInvite(Integer allianceId, String characterName) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
